packages/generator/src/generator.js

```diff
@@ -161,14 +162,22 @@ export default class Generator {
     return requireModule(path.join(this.options.buildDir, 'routes.json'))
   }
 
+  normalizeRoute (url) {
+    let pathname = decode(parseURL(url).pathname)
+    if (!this.options.router || !this.options.router.trailingSlash) {
```

Comment:
We was already decoding on L171. Any reason trainling slash handling is related to this fix? (also `route` is already of type path not full url). I was refering to [this](https://github.com/nuxt/nuxt.js/blob/fa4e5a9e1a01225b86ddda845706aa51ca277992/packages/generator/src/generator.js#L355) for nomalizing asset path and ensure is not encoded for fs